tests/api_workflow/test_api_workflow_datasources.py

```diff
@@ -142,19 +148,81 @@ def test_download_raw_sample_metadata(self):
 
     def test_download_raw_sample_metadata_progress_bar(self):
         self.api_workflow_client._datasources_api.reset()
-        pbar = mock.Mock(wraps=tqdm.tqdm(unit='file'))
+        pbar = mock.Mock(wraps=tqdm.tqdm(unit="file"))
         predictions = self.api_workflow_client.download_raw_metadata(progress_bar=pbar)
         num_samples = self.api_workflow_client._datasources_api._num_samples
         assert len(predictions) == num_samples
         pbar.update.assert_called()
 
     def test_download_raw_samples_predictions_relevant_filenames(self):
         self.api_workflow_client._datasources_api.reset()
-        predictions = self.api_workflow_client.download_raw_predictions('test', relevant_filenames_file_name="test")
+        predictions = self.api_workflow_client.download_raw_predictions(
+            "test", relevant_filenames_file_name="test"
+        )
         num_samples = self.api_workflow_client._datasources_api._num_samples
         assert len(predictions) == num_samples
 
     def test_get_prediction_read_url(self):
         self.api_workflow_client._datasources_api.reset()
-        read_url = self.api_workflow_client.get_prediction_read_url('test.json')
+        read_url = self.api_workflow_client.get_prediction_read_url("test.json")
         self.assertIsNotNone(read_url)
+
+    def test__download_raw_files_duplicate_filenames(self):
+        self.api_workflow_client._datasources_api.reset()
+        self.api_workflow_client._datasources_api._samples = defaultdict(
+            lambda: [
+                DatasourceRawSamplesDataRow(file_name="file_0", read_url="url_0"),
+                DatasourceRawSamplesDataRow(file_name="file_0", read_url="url_0"),
+                DatasourceRawSamplesDataRow(file_name="file_1", read_url="url_1"),
+                DatasourceRawSamplesDataRow(file_name="file_2", read_url="url_2"),
+                DatasourceRawSamplesDataRow(file_name="file_3", read_url="url_3"),
+                DatasourceRawSamplesDataRow(file_name="file_4", read_url="url_4"),
+            ]
+        )
+        with pytest.warns(UserWarning, match="Duplicate filename file_0 in relevant filenames file"):
+            samples = self.api_workflow_client.download_raw_samples()
+
+        assert len(samples) == 5
```

Comment:
Can you add an assertion here that the samples are the following:
```python
assert samples == [(f"file_{I}", f"url_{I}" for I in range(5)]
```

Reply:
Yes, done